oaknational/Oak-Web-Application
# [CUR-439] Refactor units tab data

## Description

Music year: 2016

- Moved useEffect in UnitsTab to [tab].tsx 
- Removed 'data' and 'examboardSlug' as props from UnitsTab
- Added trackingData prop to units tab 
- Added temporary unitTab fix for weird state changes to yearSelection in units tab (useLayoutEffect) until url routing is added to visualiser
- Updated tests & tidied up tests
Note: Passed duplicateSlugs from data formatting as a Set -> array -> Set as getStaticProps didn't like non-serializable types


## How to test

1. Go to https://deploy-preview-2411--oak-web-application.netlify.thenational.academy
2. Click on the curriculum visualiser
3. You should see no changes from the current state of the curriculum visualiser (🤞)


## Checklist
